### PR TITLE
Database migration

### PR DESCRIPTION
**Pull request by mfranciszkiewicz**
- new module: `golem.database`
- model list in `golem.model` is generated dynamically
- schema migration script generation
- schema 10 and 11 migration scripts

### CODE REVIEW

File: golem/database/database.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import logging
+
+from os import path
```
Comment (etam) [inline golem/database/database.py]: `logging` and `os` are from standard library, so should be together

File: golem/database/database.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    def close(self):
+        if not self.db.is_closed():
+            self.db.close()
+
+    def get_user_version(self) -> int:
+        cursor = self.db.execute_sql('PRAGMA user_version').fetchone()
+        return int(cursor[0])
+
+    def set_user_version(self, version: int) -> None:
```
Comment (etam) [inline golem/database/database.py:43]: Is it useful to have it as a public method?

File: golem/model.py [hunk 3]
```diff
@@ -1,93 +1,35 @@
...
-class GolemSqliteDatabase(RetryOperationalError, SqliteDatabase):
-
-    def sequence_exists(self, seq):
-        raise NotImplementedError()
-
-
 # Indicates how many KnownHosts can be stored in the DB
 MAX_STORED_HOSTS = 4
+
+# TODO: migrate to golem.database
```
Comment (etam) [inline golem/model.py:25]: What's the problem with that?

File: golem/database/database.py (hunk 2 above)
Comment (mfranciszkiewicz) [reply golem/database/database.py:43]: Yes, the migration module calls this method.

File: golem/model.py (hunk 3 above)
Comment (mfranciszkiewicz) [reply golem/model.py:25]: Having both `golem.model` and `golem.database`, the former is not an obvious location for a global database object. But moving `db` requires many additional changes and I'd rather see it done in a separate branch.

**Comment by mfranciszkiewicz** [comment]
Migration scripts need to be updated due to changes in develop and #2068 (to be merged).

**Review by Elfoniok — APPROVED** [review]

**Comment by mfranciszkiewicz** [comment]
Tests are failing due to missing migration scripts. Awaiting PR: https://github.com/golemfactory/golem/pull/2077

**Pull request merged**